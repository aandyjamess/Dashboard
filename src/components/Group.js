import React from "react";
import PropTypes from "prop-types";

import { groupWith } from "ramda";

const Group = ({Component, withFn, list, props}) => (
  <div> {groupWith(withFn, list).map((group, index) => (
    <Component key={index} groupIndex={index} group={group} {...props} />)
  )} </div>
);

Group.PropTypes = {
  withFn: PropTypes.func.isRequired,
  list: PropTypes.array.isRequired,
  props: PropTypes.object
};

export default Group;
