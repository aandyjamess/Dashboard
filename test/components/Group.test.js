import React from "react";
import toJson from "enzyme-to-json";

import { render } from "enzyme";

import Group from "../../src/components/Group";

const Component = ({groupIndex, group}) => (
  <div key={groupIndex} className="group"> {
    group.map((group, index) => <div key={groupIndex + index} className="item" />)
  } </div>
)

describe("Group", () => {
  let wrapper;

  beforeEach(() => {
    const withFn = (a, b) => a == b;
    const list = [0, 0, 1, 1, 1, 2, 2]
    wrapper = render(<Group Component={Component} withFn={withFn} list={list} />);
  });

  it("renders correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("renders groups correctly", () => {
    expect(wrapper.find(".group").length).toBe(3);
  })

  it("renders sub groups correctly", () => {
    expect(wrapper.find(".item").length).toBe(7);
  })
});
