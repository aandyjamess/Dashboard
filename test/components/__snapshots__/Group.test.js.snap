// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Group renders correctly 1`] = `
<div>
   
  <div
    class="group"
  >
     
    <div
      class="item"
    />
    <div
      class="item"
    />
     
  </div>
  <div
    class="group"
  >
     
    <div
      class="item"
    />
    <div
      class="item"
    />
    <div
      class="item"
    />
     
  </div>
  <div
    class="group"
  >
     
    <div
      class="item"
    />
    <div
      class="item"
    />
     
  </div>
   
</div>
`;
